app/exporters/contests/contest_exporter.rb

```diff
@@ -0,0 +1,100 @@
+require 'base_exporter'
+require 'csv'
+
+module Contests
+  class ContestExporter < Exporters::BaseExporter
+    include RenderAnywhere
+
+    def to_csv(contest, include_unranked_contestants)
+      problems = contest.problem_set.problems
+
+      scoreboard = CSV.generate do |csv|
+        csv << ["Rank", "User ID", "Real Name", "Username", "School", "Year"] + problems.map{ |p| p.name } + ["Total Score", "Time"]
+
+        num_ranked = 0
+        rank = nil
+        previous_record = nil
+
+        contest.scoreboard.each do |record|
+          is_ranked = !contest.only_rank_official_contestants || (record.school && record.school_year)
+          next unless is_ranked || include_unranked_contestants
+
+          if is_ranked
+            num_ranked += 1
+            rank = num_ranked unless previous_record && previous_record.score == record.score && previous_record.time_taken == record.time_taken
+            previous_record = record
+          end
```

Comment:
This complex logic is the same as in app/views/contests/scoreboard.html.erb, it should really be refactored.
But it can wait for separate PR in the interest of moving forward with other things.